### PR TITLE
Rename RolePrivilegesToApply -> GrantedRolesChange

### PR DESCRIPTION
**Pull request by matriv**
New name indicated much better the purpose of the class.

Follows: #15249

### CODE REVIEW

File: server/src/main/java/io/crate/role/GrantedRolesChange.java [hunk 1]
```diff
@@ -29,19 +29,19 @@
 import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.io.stream.Writeable;
 
-public class RolePrivilegeToApply implements Writeable {
+public class GrantedRolesChange implements Writeable {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/role/GrantedRolesChange.java:32]: Would `GrantsChange` or `RolesChange` not also communicate the same information?

**Review by mfussenegger — APPROVED** [review]
Left two other name suggestions, but `GrantedRolesChange` is also fine.

File: server/src/main/java/io/crate/role/GrantedRolesChange.java (hunk 1 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/role/GrantedRolesChange.java:32]: On second thought, `RolesChange` would be ambigous and could refer to changing a dedicated roles.

Comment (matriv) [reply server/src/main/java/io/crate/role/GrantedRolesChange.java:32]: `GrantsChange`, can also refer to simple `Privilege`, so I prefer the `GrantedRolesChange` to be honest. `RolesChange` is closer, but again (just from the name itself) could refer to changing properties of the Roles.

**Pull request merged**